### DISCUSSION TITLE
+-10v analog control

### DISCUSSION THREAD

**Discussion by imechura** [grblHAL/core#546] [General]
Does anyone know of a board that supports +-10V analog output to drives for GRBLHal. Is this supported by grblhal or is support limited to step/dir control?

**Comment by malcolmmp** [comment]
What are you looking to drive? Do you need +/-10V to motors/drivers for motion control?

If it's for standard spindle control (0-10V, positive only with separate spindle direction pin), Phil Barrett's boards are fantastic:
https://github.com/phil-barrett/grblHAL-teensy-4.x
and
https://github.com/phil-barrett/PicoCNC

I haven't heard of anything built on grbl(HAL) that'll do +/-10 for anything. Hopefully someone else can chime in.

Depending on your needs, you might require an additional circuit for logic conversion.

**Reply by terjeio** [reply]
I believe @imechura means axis control (galvanometer?) since step/dir control is mentioned.

**Comment by terjeio** [comment]
A while back I added XY axis control via an I2S analog audio DAC to a STM32F4xx/Nucleo 64 board combo. Code is [here](https://github.com/grblHAL/STM32F4xx/blob/master/boards/st_morpo_dac.c) - it traps the stepper pulse start call and scales and outputs the step position via DMA. The DAC output can be amplified/shifted to any range with suitable electronics.

**Reply by terjeio** [reply]
More [here](https://github.com/terjeio/grblHAL/discussions/138).

**Reply by iLoveAndyBaker** [reply]
Saw this and support the idea for galvo control for a laser engraver. Worth pointing out that while the ESP32 has two onboard DACs, they're only 8 bit.
